### PR TITLE
Bug 1908960: Gherkin Scripts for epic ODC-4791

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4923

Problem: Gherkin Scripts for scenarios for [ODC-4791](https://issues.redhat.com/browse/ODC-4791)

Solution: These are the gherkin scripts designed covering all the scenarios related to epics [ODC-4791](https://issues.redhat.com/browse/ODC-4791)

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 1]
```diff
@@ -83,16 +83,28 @@ Scenario: Install OpenShift Virtualization Operator: VM-01-TC01
 
 
 @regression, @smoke
-Scenario: Create HyperConverged Cluster: VM-01-TC02
+Scenario: Create OpenShift Virtuliaztion Deployment
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/operators.feature:86]: Is this scenario not related to VM-01-TC02?

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 2]
```diff
@@ -83,16 +83,28 @@ Scenario: Install OpenShift Virtualization Operator: VM-01-TC01
...
     Given user has selected namespace "openshift-cnv"
     And user is at Installed Operator page
     When user clicks on OpenShift Virtualization Operator
-    And user clicks on CNV Operator Deployment tab
-    And user clicks on the Create HyperConverged Cluster button
+    And user clicks on OpenShift Virtuliaztion Deployment tab
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/operators.feature:90]:
```suggestion
    And user clicks on OpenShift Virtualization Deployment tab
```

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 3]
```diff
@@ -83,16 +83,28 @@ Scenario: Install OpenShift Virtualization Operator: VM-01-TC01
 
 
 @regression, @smoke
-Scenario: Create HyperConverged Cluster: VM-01-TC02
+Scenario: Create OpenShift Virtuliaztion Deployment
     Given user has selected namespace "openshift-cnv"
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/operators.feature:87]: namespace "openshift-cnv' cannot be created, so you can update it as "user installed virtualization operator"

File: frontend/packages/dev-console/integration-tests/features/virtulization/create-VM-from-Template.feature [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    @smoke
+    Scenario: Create Virtual Machine
+        Given user has created Virtual Machine Template
+        And user is at Virtual Machines page
+        When user clicks on "vmware-test-template" template card
+        And user clicks on Create from template button
+        And user clicks on Create Virtual Machine button
+        Then user will be redirected to Topology page
+        And And user will see virtual machine created
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/virtulization/create-VM-from-Template.feature:21]: Add last line

Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/create-VM-from-Template.feature]: updated

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature]: updated

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature]: updated

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature]: updated

File: frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature [hunk 5]
```diff
@@ -0,0 +1,28 @@
+Feature: Create Virtual Machine templates
+    As a user, I should be able create virutal machine template to create virtual machine
+
+
+    Background:
+        Given user has installed OpenShift Virtualization Operator
+        And user has created the OpenShift Virtualization Deployment
+        And user has created the HostPathProvisioner Deployment
+        And user has selected namespace "aut-virtualization"
+        And user is at administrator perspective
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature:10]:
```suggestion
        And user is at administrator perspective
        And user has selected namespace "aut-virtualization"
```

Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature [hunk 6]
```diff
@@ -2,91 +2,91 @@ Feature: Perform Actions on created VM
...
-@regression
-Scenario: Delete VM action on VM: VM-04-TC08
-    When user right clicks on the VM to open the context menu
-    And user clicks on the Delete VM
-    And user clicks on the Delete button on the modal
-    Then VM will get deleted
+    Background:
+        Given user is at developer perspecitve
+        And user has selected namespace "aut-vm-actions"
+        And user has created VM
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature:8]: mention the VM name

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature [hunk 7]
```diff
@@ -2,91 +2,91 @@ Feature: Perform Actions on created VM
...
+        And user is at the Topolgy page
+
+
+    @regression
+    Scenario: Sidebar for VM: VM-03
+        When user clicks on the VM to open the sidebar
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature:14]: Update the test data like "user clicks on the VM "vm_name" to open the sidebar"

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature [hunk 8]
```diff
@@ -2,91 +2,91 @@ Feature: Perform Actions on created VM
...
+        Then user can see the Details tab
+        And user can see the Resources tab
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature:16]:
```suggestion
        Then user can see the Details tab, Resources tab
```

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature [hunk 9]
```diff
@@ -2,91 +2,91 @@ Feature: Perform Actions on created VM
...
+        And user can see the Actions dropdown
+
+
+    @regression
+    Scenario: Edit Application Groupings action on VM: VM-04-TC02
+        When user right clicks on the VM to open the context menu
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature:22]: test data required

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature [hunk 10]
```diff
@@ -2,91 +2,91 @@ Feature: Perform Actions on created VM
...
+        And user clicks on the Edit Application Groupings
+        And user will click on the Application dropdown on the modal
+        And user selects the Application
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature:25]: user selects the Application "app_name"

**Review by makambalaji — CHANGES_REQUESTED** [review]
Please add the test data, wherever applicable and add the @cnv tag to all the vm scenarios

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature (hunk 7 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature (hunk 8 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature (hunk 10 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature (hunk 9 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/actions-on-Vm.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-page.feature [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    Scenario: Virtual Machines on Developer Catalog page
+        Given user is at Developer Catalog page
+        Then user will see Virtual Machines type
+    
+
+    @smoke, @cnv
+    Scenario: Virtual Machines page - Empty View
+        Given user is at Developer Catalog page
+        When user clicks on Virtual Machines type
+        Then user will see "No Catalog items found"
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-page.feature:29]: If the user has created OpenShift Virtualization Deployment and HostPathProvisioner Deployment how can it be empty

File: frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        And user has created the OpenShift Virtualization Deployment
+        And user has created the HostPathProvisioner Deployment
+        And user is at administrator perspective
+        And user has selected namespace "aut-virtualization"
+        
+
+
+    @smoke, @cnv, @regression
+    Scenario: Create Virutal Machine template
+        Given user is at Virtualization tab under Workloads nav-item
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature:16]: I think this is missing: user is at the administration view

File: frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+        And user enters name "vmware-test-template"
+        And user selects "Fedora 31 or higher" from Operating System dropdown
+        And user selects "Import via Registry (creates PVC)" from Boot Source dropdown
+        And user enters "kubevirt/fedora-cloud-container-disk-demo" in Container Image field
+        And user selects "Small - 1 vCPU, 2 GiB Memory" from Flavor dropdown
+        And user selects "Desktop" from Workload Type dropdown
+        And user clicks on Review and Confirm button
+        And user clicks on Create Virtual Machine template
+        Then user will see "Successfully created virtual machine template" message
+        And user will see See virtual machine template details button
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature:29]: Can we add something like topology will have a vm workload

File: frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-page.feature (hunk 11 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-page.feature:29]: Creating these two deployments won't create any templates

File: frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature (hunk 12 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/virtulization/virtualMachines-template.feature:16]: It's in the background

**Comment by andrewballantyne** [comment]
/cc @christianvogt 

Christian worked more with the CNV team than I did.
